registration/models.py

```diff
@@ -376,3 +377,364 @@ def send_activation_email(self, site, request=None):
+
+        The email will use the following templates,
+        which can be overriden by settings APPROVAL_COMPLETE_EMAIL_SUBJECT,
+        APPROVAL_COMPLETE_EMAIL_BODY, and APPROVAL_COMPLETE_EMAIL_HTML appropriately:
+
+        ``registration/admin_approve_complete_email_subject.txt``
+            This template will be used for the subject line of the
+            email. Because it is used as the subject line of an email,
+            this template's output **must** be only a single line of
+            text; output longer than one line will be forcibly joined
+            into only a single line.
+
+        ``registration/admin_approve_complete_email.txt``
+            This template will be used for the text body of the email.
+
+        ``registration/admin_approve_complete_email.html``
+            This template will be used for the text body of the email.
+
+        These templates will each receive the following context
+        variables:
+
+        ``user``
+            The new user account
+
+        ``site``
+            An object representing the site on which the user
+            registered; depending on whether ``django.contrib.sites``
+            is installed, this may be an instance of either
+            ``django.contrib.sites.models.Site`` (if the sites
+            application is installed) or
+            ``django.contrib.sites.requests.RequestSite`` (if
+            not). Consult the documentation for the Django sites
+            framework for details regarding these objects' interfaces.
+
+        ``request``
+            Optional Django's ``HttpRequest`` object from view.
+            If supplied will be passed to the template for better
+            flexibility via ``RequestContext``.
+        """
+        admin_approve_complete_email_subject = getattr(
+            settings, 'APPROVAL_COMPLETE_EMAIL_SUBJECT',
+            'registration/admin_approve_complete_email_subject.txt')
+        admin_approve_complete_email_body = getattr(
+            settings, 'APPROVAL_COMPLETE_EMAIL_BODY',
+            'registration/admin_approve_complete_email.txt')
+        admin_approve_complete_email_html = getattr(
+            settings, 'APPROVAL_COMPLETE_EMAIL_HTML',
+            'registration/admin_approve_complete_email.html')
+
+        ctx_dict = {}
+        if request is not None:
+            ctx_dict = RequestContext(request, ctx_dict)
+        # update ctx_dict after RequestContext is created
+        # because template context processors
+        # can overwrite some of the values like user
+        # if django.contrib.auth.context_processors.auth is used
+        ctx_dict.update({
+            'user': self.user,
+            'site': site,
+        })
+        subject = (getattr(settings, 'REGISTRATION_EMAIL_SUBJECT_PREFIX', '') +
+                   render_to_string(
+                       admin_approve_complete_email_subject, ctx_dict))
+        # Email subject *must not* contain newlines
```

Comment:
can you abstract this as a send_email helper function? this code is repeated in the `SupervisedRegistrationManager` and `RegistrationProfile` classes